### PR TITLE
fix 6892 - avoid adding clause when table not present

### LINKED ISSUE #6892: [Bug][GitLab] raw table is missing when collecting api trigger jobs
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

In #6167, `CollectApiTriggerJobs` was added, but in function `GetAllPipelinesIterator`, `_raw_gitlab_api_trigger_job` is used before this table was created. 
And when I running a project with a gitlab connection, some errors happened.

```
time="2024-01-29 19:39:08" level=info msg=" [pipeline service] [pipeline #8] [task #59] executing subtask collectApiTriggerJobs"
time="2024-01-29 19:39:08" level=info msg="\x1b[31;1m/Users/houlinwei/Code/go/src/github.com/merico-dev/lake-feat-6854/backend/impls/dalgorm/dalgorm.go:188 \x1b[35;1mError 1146 (42S02): Table 'lake._raw_gitlab_api_trigger_job' doesn't exist\n\x1b[0m\x1b[33m[2.624ms] \x1b[34;1m[rows:-]\x1b[0m SELECT gp.gitlab_id, gp.gitlab_id as iid FROM _tool_gitlab_pipelines gp WHERE gp.project_id = 50751699 and gp.connection_id = 1 and gp.gitlab_id not in (select json_extract(tj.input, '$.GitlabId') as gitlab_id from _raw_gitlab_api_trigger_job tj)"
time="2024-01-29 19:39:08" level=error msg=" [pipeline service] [pipeline #8] [task #59] subtask collectApiTriggerJobs ended unexpectedly\n\tWraps: (2) Error 1146 (42S02): Table 'lake._raw_gitlab_api_trigger_job' doesn't exist (500)\n\tWraps: (3) Error 1146 (42S02): Table 'lake._raw_gitlab_api_trigger_job' doesn't exist\n\tError types: (1) *hintdetail.withDetail (2) *hintdetail.withDetail (3) *mysql.MySQLError"
```

### What do you expect to happen

Run project sucessfully.

### How to reproduce

Create a project with a gitlab connection, collect data.

### Anything else

_No response_

### Version

main

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Marco-De-Stefani**
### Summary
Fixing #6892 
by removing the clause if the table does not exists

### Does this close any open issues?
Closes #6892

### CODE REVIEW

**Comment by d4x1** [comment]
LGTM.

File: backend/plugins/gitlab/tasks/trigger_job_collector.go [hunk 1]
```diff
@@ -83,10 +83,14 @@ func GetAllPipelinesIterator(taskCtx plugin.SubTaskContext, collectorWithState *
...
 		dal.From("_tool_gitlab_pipelines gp"),
 		dal.Where(
-			`gp.project_id = ? and gp.connection_id = ? and gp.gitlab_id not in (select json_extract(tj.input, '$.GitlabId') as gitlab_id from _raw_gitlab_api_trigger_job tj)`,
+			`gp.project_id = ? and gp.connection_id = ? `,
 			data.Options.ProjectId, data.Options.ConnectionId,
 		),
 	}
+
+	if db.HasTable("_raw_gitlab_api_trigger_job") {
+		clauses = append(clauses, dal.Where("and gp.gitlab_id not in (select json_extract(tj.input, '$.GitlabId') as gitlab_id from _raw_gitlab_api_trigger_job tj)"))
```
Comment (klesh) [inline backend/plugins/gitlab/tasks/trigger_job_collector.go:92]: Please remove the `and` at the beginning of the sql

**Review by klesh — DISMISSED** [review]

**Review by d4x1 — DISMISSED** [review]

**Review by d4x1 — APPROVED** [review]

**Comment by meeroslaph** [comment]
Is this fix going to be included into any release?

**Comment by d4x1** [comment]
@meeroslaph  It's included in https://github.com/apache/incubator-devlake/releases/tag/v1.0.1-beta6.

**Pull request merged**